src/StructuredLogger/Strings/Strings.cs

```diff
@@ -344,14 +344,29 @@ private static string GetPropertyReassignmentText()
             return "^" + text + "$";
         }
 
-        public static Regex CreateRegex(string text, int replacePlaceholders = 0, RegexOptions options = RegexOptions.Compiled)
+        public static Regex CreateRegex(string text, int replacePlaceholders = 0, RegexOptions options = RegexOptions.Compiled | RegexOptions.Singleline, bool capture = false)
         {
-            text = Regex.Escape(text);
+            if (text.StartsWith("^") || !text.EndsWith("$"))

```

Comment:
I don't quite understand this condition. Can you double-check?

I'd write this as:
```csharp
text = Regex.Escape(text);
if (!text.StartsWith("^") && !text.EndsWith("$"))
{
    text = $"^{text}$";
}
```

Reply:
I haven't followed the conversation here, but I thought I'd point out that `text.StartsWith("^")` will always be false if `text` is an escaped regex.